src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -75,6 +76,12 @@ def load_arguments(self, _):
         c.argument('no_self_perms', arg_type=get_three_state_flag(), help="Don't add permissions for the current user/service principal in the new vault.")
         c.argument('location', validator=get_default_location_from_resource_group)
 
+    for item in ['recover', 'purge']:
+        with self.argument_context('keyvault ' + item) as c:
+            c.argument('vault_name', help='Name of the deleted vault', required=True, completer=None, validator=validate_deleted_vault_name)
+            c.argument('resource_group_name', resource_group_name_type, id_part=None, required=False, help='Resource group of the deleted vault.')
+            c.argument('location', help='Location of the deleted vault to be restored.', required=False, validator=get_default_location_from_resource_group)
```

Comment:
Do you need the explicit `required=True/False`?  Generally, this is inferred by whether a default is specified in the command signature.

Reply:
The required=False on resource_group_name and location is because those are actually required parameters as far as the method is concerned.  However the validator will add them if they aren't specified so it's not required the users specify them.